crates/amaru-consensus/src/consensus/stages/select_chain.rs

```diff
@@ -148,19 +178,31 @@ impl SelectChain {
         chain_sync: DecodedChainSyncEvent,
     ) -> BoxFuture<'_, Result<Vec<ValidateHeaderEvent>, ConsensusError>> {
         Box::pin(async move {
-            match chain_sync {
+            let ret = match chain_sync {
                 DecodedChainSyncEvent::RollForward {
                     peer, header, span, ..
-                } => self.select_chain(store, peer, header, span).await,
+                } => self.select_chain(store.clone(), peer, header, span).await,
                 DecodedChainSyncEvent::Rollback {
                     peer,
                     rollback_point,
                     span,
                 } => {
-                    self.select_rollback(store, peer, rollback_point, span)
+                    self.select_rollback(store.clone(), peer, rollback_point, span)
                         .await
                 }
+            };
+
+            if ret.as_ref().map(|evs| !evs.is_empty()).unwrap_or(false) {
+                // FIXME add some more points in between?
+                let hashes = [store.get_anchor_hash(), store.get_best_chain_hash()];
+                let points = hashes
+                    .iter()
+                    .filter_map(|h| store.load_header(h).map(|h| h.point()))
+                    .collect();
+                *self.intersection.write() = points;
```

Comment:
@abailly This is the place where chain updates are communicated to the upstream stages for better intersection after reconnect.

Reply:
Would it not be better to use the `ReadOnlyChainStore` for that? Seems to me this would make the overall interactions simpler to understand